### ISSUE TITLE
Warn admin about outdated Collabora versions

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/richdocuments#4349]
We notice quite often that users update their Nextcloud but have no information that there is a new Collabora version available (both patch updates but also major versions).

On the Nextcloud side we can send notifications to admins and also warn them prominently in the admin overview.

While Collabora has a hint about new versions it is only shown to admins when they use collabora which might not happen often and therefore might be unseen.

My proposal would be to extend this in the richdocuments app:

- Regularly check if there is a new Collabora version available
- Notify admins about it
- Show a warning in the admin overview of Nextcloud

We can read the currently installed version from capabilities of Collabora but there is no way I'm aware to know about the latest published version in machine readable form (both for CODE and COOL releases).

@pedropintosilva Not sure who to ping, but it would be nice to just have some endpoint that we can query that will tell us the latest version tag for CODE / COOL.

**Comment by pedropintosilva** [comment]
I agree. I wonder if this could be integral part of the https://github.com/nextcloud/richdocuments/issues/4041

**Comment by pedropintosilva** [comment]
Nevertheless, I wonder if we don't already have a url that gives us that or if it wouldn't be beneficial to have  one but probably @timar knows best

**Comment by juliusknorr** [comment]
> I agree. I wonder if this could be integral part of the https://github.com/nextcloud/richdocuments/issues/4041

Not sure how this should be related

**Comment by juliusknorr** [comment]
For CODE I could imagine we just fetch https://www.collaboraoffice.com/repos/CollaboraOnline/CODE-deb/Packages and get the coolwsd version. But for enterprise packages it would not be feasible to use a real repository URL.

@timar A possibly simple solution I imagine might be to just write a simple text file containing the version during the build and just put them somewhere next to the repos. Maybe that can even be automated in the build scripts (I assume you have some in place for that 😁 )

Then we could hardcode urls like this in the richdocuments app to check for the latest release and compare with the ones that online returns in the capabilities

```
https://www.collaboraoffice.com/repos/latest-version-code
https://www.collaboraoffice.com/repos/latest-version-cool

**Comment by juliusknorr** [comment]
We can use the following two:

```
For CODE: https://rating.collaboraonline.com/UpdateCheck/updatecheck2.html
For COOL: https://rating.collaboraonline.com/UpdateCheck/updatecheck2.html?product=cool

**Comment by timar** [comment]
> We can use the following two:
> 
> ```
> For CODE: https://rating.collaboraonline.com/UpdateCheck/updatecheck2.html
> For COOL: https://rating.collaboraonline.com/UpdateCheck/updatecheck2.html?product=cool
> ```

Yes, indeed. In fact the URLs above are OK, but the logic is a bit different. When the `product` parameter is empty, the code assumes CODE (i.e. product=Collabora%20Online%20Development%20Edition). Otherwise endpoint returns COOL version, whatever else you pass with the `product` parameter. The `/hosting/capabilities` endpoint of COOL server returns the product name in the `productName` property. You can use it to query the update check service.
